pkg/apis/pipeline/v1alpha1/task_validation.go

```diff
@@ -64,6 +65,13 @@ func (ts *TaskSpec) Validate() *apis.FieldError {
 			return err
 		}
 	}
+
+	// Validate task step names
+	for _, step := range ts.Steps {
+		if errs := validation.IsDNS1123Label(step.Name); len(errs) > 0 {
+			return apis.ErrInvalidValue("taskspec.steps.name", step.Name)
```

Comment:
what info will the user see when this is hit? if we could possibly provide them with info about what they need to comply with that would be great (maybe that's already in `errs`?)

Reply:
Yes `errs` contain the validation error. Following is the error msg msg for invalid task name.

```
Error from server (InternalError): error when creating "examples/deploy_tasks.yaml": Internal error occurred: admission webhook "webhook.pipeline.knative.dev" denied the request: mutation failed: invalid value "taskspec.steps.name": replaceRedisImage
```

> if we could possibly provide them with info about what they need to comply with 

If we want to pass the error to the user then probably build field error object with `details` key to include this information. WDYT? @bobcatfish

Reply:
that sounds great to me @shashwathi !

Reply:
I have updated the error to include k8s documentation link about names convention. This is the new error

```
invalid value "replaceImage": taskspec.steps.name
Task step name must be a valid DNS Label, For more info refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
```